provider/cloudflare/cloudflare_test.go

```diff
@@ -3618,3 +3533,794 @@ func TestZoneService(t *testing.T) {
+		assert.Equal(t, "origin.example.com", hostnames[0].CustomOriginServer)
+		assert.Equal(t, "sni.example.com", hostnames[0].CustomOriginSNI)
+	})
+
+	t.Run("MultipleCustomHostnames", func(t *testing.T) {
+		mockHostnames := []custom_hostnames.CustomHostnameListResponse{
+			{
+				ID:                 "ch1",
+				Hostname:           "test1.example.com",
+				CustomOriginServer: "origin1.example.com",
+				CustomOriginSNI:    "sni1.example.com",
+			},
+			{
+				ID:                 "ch2",
+				Hostname:           "test2.example.com",
+				CustomOriginServer: "origin2.example.com",
+				CustomOriginSNI:    "sni2.example.com",
+			},
+			{
+				ID:                 "ch3",
+				Hostname:           "test3.example.com",
+				CustomOriginServer: "origin3.example.com",
+				CustomOriginSNI:    "sni3.example.com",
+			},
+		}
+		pager := &mockAutoPager[custom_hostnames.CustomHostnameListResponse]{
+			items: mockHostnames,
+		}
+
+		hostnames, err := listAllCustomHostnames(pager)
+
+		assert.NoError(t, err)
+		assert.Len(t, hostnames, 3)
+
+		// Verify all hostnames are correctly converted
+		for i, hostname := range hostnames {
+			expected := mockHostnames[i]
+			assert.Equal(t, expected.ID, hostname.ID)
+			assert.Equal(t, expected.Hostname, hostname.Hostname)
+			assert.Equal(t, expected.CustomOriginServer, hostname.CustomOriginServer)
+			assert.Equal(t, expected.CustomOriginSNI, hostname.CustomOriginSNI)
+		}
+	})
+
+	t.Run("IteratorError", func(t *testing.T) {
+		expectedErr := errors.New("iterator error")
+		mockHostname := custom_hostnames.CustomHostnameListResponse{
+			ID:                 "ch1",
+			Hostname:           "test.example.com",
+			CustomOriginServer: "origin.example.com",
+		}
+		pager := &mockAutoPager[custom_hostnames.CustomHostnameListResponse]{
+			items:    []custom_hostnames.CustomHostnameListResponse{mockHostname},
+			err:      expectedErr,
+			errIndex: 1, // Error after first item
+		}
+
+		hostnames, err := listAllCustomHostnames(pager)
+
+		assert.Error(t, err)
+		assert.Equal(t, expectedErr, err)
+		assert.Empty(t, hostnames) // Should be empty on error
+	})
+
+	t.Run("PartialIteratorError", func(t *testing.T) {
+		expectedErr := errors.New("partial iterator error")
+		mockHostnames := []custom_hostnames.CustomHostnameListResponse{
+			{
+				ID:                 "ch1",
+				Hostname:           "test1.example.com",
+				CustomOriginServer: "origin1.example.com",
+			},
+			{
+				ID:                 "ch2",
+				Hostname:           "test2.example.com",
+				CustomOriginServer: "origin2.example.com",
+			},
+		}
+		pager := &mockAutoPager[custom_hostnames.CustomHostnameListResponse]{
+			items:    mockHostnames,
+			err:      expectedErr,
+			errIndex: 2, // Error after second item
+		}
+
+		hostnames, err := listAllCustomHostnames(pager)
+
+		assert.Error(t, err)
+		assert.Equal(t, expectedErr, err)
+		assert.Empty(t, hostnames) // Should be empty on error
+	})
+}
+
+func TestBuildCustomHostnameSSLParams(t *testing.T) {
```

Comment:
Not sure what this is supposed to test 😄.
This doesn't call any func from this package.